params/json.libevm.go

```diff
@@ -47,13 +47,13 @@ func (c *ChainConfig) UnmarshalJSON(data []byte) error {
 
 	case reg.Registered() && reg.Get().reuseJSONRoot: // although the latter is redundant, it's clearer
 		c.extra = reg.Get().newChainConfig()
-		if err := json.Unmarshal(data, c.extra); err != nil {
+		if err := UnmarshalChainConfigJSON(data, nil, c.extra); err != nil {
 			c.extra = nil
 			return err
 		}
 		fallthrough // Important! We've only unmarshalled the extra field.
 	default: // reg == nil
-		return json.Unmarshal(data, (*chainConfigWithoutMethods)(c))
+		return UnmarshalChainConfigJSON[struct{}](data, c, nil)
```

Comment:
@ARR4N I'm not sure this is what you were suggesting? Or moving this switch logic within `UnmarshalChainConfigJSON`?